### PR TITLE
Documentation: Make the SerenityOS Patterns documentation more idiomatic

### PR DESCRIPTION
**Pull request by ninadsachania**

### CODE REVIEW

File: Documentation/Patterns.md [hunk 1]
```diff
@@ -164,6 +165,7 @@ Instead, `serenity_main(..)` is defined like this:
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
+    // code
```
Comment (nico) [inline Documentation/Patterns.md:168]: Same.

File: Documentation/Patterns.md [hunk 2]
```diff
@@ -3,22 +3,22 @@
...
-Any code surrounded by `TRY(..)` will attempt to be executed, and any error
+Any code surrounded by `TRY(...)` will attempt to be executed, and any error
 will immediately be returned from the function. If no error occurs then the
-result of the contents of the TRY will be the result of the macro's execution.
+result of the contents of the `TRY(...)` will be the result of the macro's execution.
 
 ### Examples:
 
-Example from LibGfx:
+Example from [LibGfx](../Userland/Libraries/LibGfx/Bitmap.cpp):
```
Comment (nico) [inline Documentation/Patterns.md:21]: Let's not link to actual code. It adds close to nothing, and adds the risk of the link becoming stale.

File: Documentation/Patterns.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ ErrorOr<NonnullRefPtr<Bitmap>> Bitmap::create_shareable(BitmapFormat format, Int
 }
 ```
 
-Example from the Kernel:
+Example from the [Kernel](../Kernel/Memory/AddressSpace.cpp):
```
Comment (nico) [inline Documentation/Patterns.md:42]: same

File: Documentation/Patterns.md [hunk 4]
```diff
@@ -130,30 +130,31 @@ private:
...
 ### Examples:
 
-A function `main(..)` would normally look something like:
+In C and C++, the main function normally looks something like:
 
 ```cpp
 int main(int argc, char** argv)
 {
-   return 0;
+    // code
```
Comment (nico) [inline Documentation/Patterns.md:156]: This line adds nothing, remove it.

File: Documentation/Patterns.md [hunk 5]
```diff
@@ -222,18 +224,18 @@ class MemoryManager {
...
+forces one to go investigate by printing out the size, checking it in a
 debugger, etc.
 
 For this reason `AK::AssertSize` was added. It exploits the fact that the
 compiler will emit template argument values for compiler errors to provide
 debugging information. Instead of getting no information you'll get the actual
 type sizes in your compiler error output.
 
-Example Usage:
+Example usage:
```
Comment (nico) [inline Documentation/Patterns.md:238]: I think LHS is supposed to be title case. Leave these alone.

File: Documentation/Patterns.md [hunk 6]
```diff
@@ -303,23 +304,23 @@ static StringView example_fn(const SourceLocation& loc = SourceLocation::current
     return loc.function_name();
 }
 
-int main(int, char**)
+int main(void)
```
Comment (nico) [inline Documentation/Patterns.md:307]: Undo this change. (writing an explicit `void` in a parameter list is C, where functions without any parameters are implicitly varargs. In C++, it's `main()`, but the LHS is fine too. The RHS is wrong.)

**Review by nico — COMMENTED** [review]
Borders on frivolous, but I suppose some of these are kinda better. Others are worse though, and the sum total benefit has a hard time to be worth the time spent on it. Againm, I'd recommend working on less superficial things.

(Still, I looked at it, and if the next revision addresses comments, I'll hit merge.)

File: Documentation/Patterns.md (hunk 2 above)
Comment (ninadsachania) [reply Documentation/Patterns.md:21]: Done.

**Comment by ninadsachania** [comment]
> Borders on frivolous, but I suppose some of these are kinda better. Others are worse though, and the sum total benefit has a hard time to be worth the time spent on it. Again, I'd recommend working on more substantive things.
> 
> (Still, I looked at it, and if the next revision addresses comments, I'll hit merge.)

Thanks for reviewing!

I did create PRs with more substantive improvements, but nobody is reviewing them.

Here are some examples:

+ https://github.com/SerenityOS/serenity/pull/25435
+ https://github.com/SerenityOS/serenity/pull/25405
+ https://github.com/SerenityOS/serenity/pull/25388

**Comment by nico** [comment]
> I did create PRs with more substantive improvements, but nobody is reviewing them.

They are also < 10 lines of code touched – _even more_ substantive! (I felt like I said this last time too, and then kind of disengaged when the message didn't seem to be heard.)

**Comment by ninadsachania** [comment]
> > I did create PRs with more substantive improvements, but nobody is reviewing them.
> 
> They are also < 10 lines of code touched – _even more_ substantive! (I felt like I said this last time too, and then kind of disengaged when the message didn't seem to be heard.)

I am confused by what you mean by "substantive"? Should I create PRs that touch lots of lines?

Two of PRs that I linked above are bug fixes. I don't know how to make them more substantive than that.

File: Documentation/Patterns.md [hunk 7]
```diff
@@ -319,7 +318,7 @@ Example Usage:
 ```cpp
 
 #if LOCK_DEBUG
-#    include <AK/SourceLocation.h>
+    #include <AK/SourceLocation.h>
```
[the remark this answers is not in the document]
Comment (ninadsachania) [reply Documentation/Patterns.md]: No, it's now new. That was a mistake on my part.

**Pull request merged**